lib/src/facts/freebsd/dmi_resolver.cc

```diff
@@ -0,0 +1,36 @@
+#include <internal/facts/freebsd/dmi_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+
+#include <kenv.h>
+
+using namespace std;
+
+namespace facter { namespace facts { namespace freebsd {
+
+    dmi_resolver::data dmi_resolver::collect_data(collection& facts)
+    {
+        data result;
+        result.bios_vendor = kenv_lookup("smbios.bios.vendor");
+        result.bios_version = kenv_lookup("smbios.bios.version");
+        result.bios_release_date = kenv_lookup("smbios.bios.reldate");
+        result.uuid = kenv_lookup("smbios.system.uuid");
+        result.serial_number = kenv_lookup("smbios.planar.serial");
+        result.product_name = kenv_lookup("smbios.system.product");
+        result.manufacturer = kenv_lookup("smbios.system.maker");
+
+        return result;
+    }
+
+    string dmi_resolver::kenv_lookup(const char* file)
+    {
+        char buffer[100] = {};
+
+        LOG_DEBUG("localchassis", "DMI request for %s", file);
+        if (kenv(KENV_GET, file, buffer, sizeof(buffer) - 1) == -1) {
+            LOG_DEBUG("localchassis", "cannot get %s", file);
```

Comment:
Yep, fixed it up to the similar solaris code, I edited the lookup to a deliberately missing kenv setting to check the error:

```
2016-06-12 17:47:01.032786 DEBUG puppetlabs.facter - kenv lookup for smbios.system.failme
2016-06-12 17:47:01.032870 WARN  puppetlabs.facter - kenv lookup for smbios.system.failme failed: No such file or directory (2)
```